Kernel/Storage/NVME/NVMENameSpace.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021, Pankaj Raghav <pankydev8@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#include "NVMENameSpace.h"
+#include <AK/NonnullOwnPtr.h>
+#include <Kernel/Storage/NVME/NVMENameSpace.h>
+#include <Kernel/Storage/StorageManagement.h>
+
+namespace Kernel {
+
+NonnullRefPtr<NVMENameSpace> NVMENameSpace::create(NonnullRefPtrVector<NVMEQueue> queues, u16 nsid, size_t storage_size, size_t lba_size)
+{
+    auto minor_number = StorageManagement::minor_number();
+    auto major_number = StorageManagement::major_number();
+    // TODO: For now we are assuming only one controller is there. Name format nvme<ctrlid>n<nsid>
+    auto device_name = String::formatted("nvme0n{:d}", nsid);
+    auto device_name_kstring = KString::must_create(device_name.view());
+    return adopt_ref(*new NVMENameSpace(queues, storage_size, lba_size, major_number, minor_number, nsid, move(device_name_kstring)));
```

Comment:
This is not the correct way to create a block/character device.
You should do something like this:
```c++
    auto disk_device_or_error = DeviceManagement::try_create_device<NVMENameSpace>(...);
```
This is why the disk device doesn't appear on the `/sys/dev/block` directory.